WalletWasabi.Tests/UnitTests/WabiSabi/Client/ArenaClientTests.cs

```diff
@@ -68,75 +66,70 @@ public async Task FullCoinjoinAsyncTestAsync()
 			// Phase: Input Registration
 			Assert.Equal(Phase.InputRegistration, round.Phase);
 
-			await aliceArenaClient.ConfirmConnectionAsync(
+			var connectionConfirmationResponse1 = await aliceArenaClient.ConfirmConnectionAsync(
 				round.Id,
 				aliceId,
 				inputRemainingVsizes,
-				amountCredentials.ZeroValue.Take(ProtocolConstants.CredentialNumber),
+				inputRegistrationResponse.RealAmountCredentials,
+				inputRegistrationResponse.RealVsizeCredentials,
 				reissuanceAmounts,
 				CancellationToken.None);
 
-			Assert.Empty(amountCredentials.Valuable);
-
 			await arena.TriggerAndWaitRoundAsync(TimeSpan.FromMinutes(1));
 			Assert.Equal(Phase.ConnectionConfirmation, round.Phase);
 
 			// Phase: Connection Confirmation
-			await aliceArenaClient.ConfirmConnectionAsync(
+			var connectionConfirmationResponse2 = await aliceArenaClient.ConfirmConnectionAsync(
 				round.Id,
 				aliceId,
 				inputRemainingVsizes,
-				amountCredentials.ZeroValue.Take(ProtocolConstants.CredentialNumber),
+				connectionConfirmationResponse1.RealAmountCredentials,
+				connectionConfirmationResponse1.RealVsizeCredentials,
 				reissuanceAmounts,
 				CancellationToken.None);
 
-			Assert.Single(amountCredentials.Valuable, x => x.Amount.ToMoney() == reissuanceAmounts.First());
-			Assert.Single(amountCredentials.Valuable, x => x.Amount.ToMoney() == reissuanceAmounts.Last());
-
-			await arena.TriggerAndWaitRoundAsync(TimeSpan.FromMinutes(1));
+			await arena.TriggerAndWaitRoundAsync(TimeSpan.FromSeconds(1));
 			Assert.Equal(Phase.OutputRegistration, round.Phase);
 
-			var bobArenaClient = new ArenaClient(round.AmountCredentialIssuerParameters, round.VsizeCredentialIssuerParameters, amountCredentials, vsizeCredentials, coordinator, new InsecureRandom());
-
-			Assert.Equal(4, amountCredentials.ZeroValue.Count());
+			var bobArenaClient = new ArenaClient(round.AmountCredentialIssuerParameters, round.VsizeCredentialIssuerParameters, zeroAmountCredentials, zeroVsizeCredentials, coordinator, new InsecureRandom());
 
 			// Phase: Output Registration
 			using var destinationKey1 = new Key();
 			using var destinationKey2 = new Key();
 
-			var result = await bobArenaClient.ReissueCredentialAsync(
+			var reissuanceResponse = await bobArenaClient.ReissueCredentialAsync(
 				round.Id,
 				reissuanceAmounts[0],
 				destinationKey1.PubKey.WitHash.ScriptPubKey,
 				reissuanceAmounts[1],
 				destinationKey2.PubKey.WitHash.ScriptPubKey,
-				amountCredentials.Valuable,
-				vsizeCredentials.Valuable,
+				connectionConfirmationResponse2.RealAmountCredentials,
+				connectionConfirmationResponse2.RealVsizeCredentials,
 				CancellationToken.None);
 
-			Assert.Equal(6, amountCredentials.ZeroValue.Count());
-			Assert.Equal(6, vsizeCredentials.ZeroValue.Count());
+//			Assert.Equal(6, amountCredentials.ZeroValue.Count());
+//			Assert.Equal(6, vsizeCredentials.ZeroValue.Count());
```

Comment:
Should these get removed?

Reply:
Yes, they should be removed. This kind of checks are fragile in nature because they doesn't allow us to change the way things work without breaking the tests and at the same time they force us to expose internal state that should not be accessible.

Reply:
Done. Thanks